### PR TITLE
Add option to disable IMDS from nodegroups

### PR DESCRIPTION
**Pull request by michaelbeaumont**
### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [x] Added labels for change area (e.g. `area/nodegroup`), target version (e.g. `version/0.12.0`) and kind (e.g. `kind/improvement`)
- [x] Make sure the title of the PR is a good description that can go into the release notes

### CODE REVIEW

File: pkg/apis/eksctl.io/v1alpha5/defaults.go [hunk 1]
```diff
@@ -113,6 +113,13 @@ func setNodeGroupBaseDefaults(ng *NodeGroupBase, meta *ClusterMeta) {
 		ng.Labels = make(map[string]string)
 	}
 	setDefaultNodeLabels(ng.Labels, meta.Name, ng.Name)
+
+	if ng.DisableIMDSv1 == nil {
+		ng.DisableIMDSv1 = Disabled()
+	}
+	if ng.DisableIMDS == nil {
```
Comment (martina-if) [inline pkg/apis/eksctl.io/v1alpha5/defaults.go:120]: Should it be `DisableIMDSv2`?

File: pkg/nodebootstrap/userdata_al2.go [hunk 2]
```diff
@@ -58,6 +58,10 @@ func NewUserDataForAmazonLinux2(spec *api.ClusterConfig, ng *api.NodeGroup) (str
 		config.AddShellCommand(command)
 	}
 
+	if api.IsEnabled(ng.DisableIMDS) {
+		scripts = append(scripts, "disable-imds.al2.sh")
```
Comment (martina-if) [inline pkg/nodebootstrap/userdata_al2.go:62]: What about Ubuntu AMIs? If it won't be supported for other AMIs perhaps we need to validate that in the beginning and error out? or print a warning?

**Comment by cPu1** [comment]
We also need to support disabling IMDS for managed nodegroups. I have pushed the changes for it.

File: pkg/apis/eksctl.io/v1alpha5/types.go [hunk 3]
```diff
@@ -1094,6 +1094,11 @@ type NodeGroupBase struct {
 	// Defaults to `false`
 	// +optional
 	DisableIMDSv1 *bool `json:"disableIMDSv1,omitempty"`
+
+	// DisableIMDS disallows IMDS requests
```
Comment (cPu1) [inline pkg/apis/eksctl.io/v1alpha5/types.go:1098]: WDYT about "blocks IMDS requests"?

```suggestion
	// DisableIMDS blocks IMDS requests
```

**Comment by michaelbeaumont** [comment]
https://github.com/weaveworks/eksctl/pull/2634

**Pull request closed without merging**